crates/bootstrap_srv/src/http.rs

```diff
@@ -168,23 +213,41 @@ fn tokio_thread(
 
                 let app = app.clone();
                 let shutdown_handle = shutdown_handle.clone();
-                if let Some(tls_config) = &config.tls_config {
+                if let Some(tls_config) = &server_config.tls_config {
                     let tls_config = tls_config
                         .create_tls_config()
                         .await
                         .expect("Failed to create TLS config");
 
+                    let acceptor = RustlsAcceptor::new(tls_config);
+
+                    let acceptor = {
+                        acceptor.acceptor(crate::sbd::SbdAcceptor::new(
+                                sbd_config.clone(),
+                                ip_rate.clone(),
+                            ))
+                    };
+
                     let s = axum_server::Server::from_tcp(listener)
-                        .acceptor(RustlsAcceptor::new(tls_config))
+                        .acceptor(acceptor)
                         .handle(shutdown_handle)
-                        .serve(app.into_make_service());
+                        .serve(app.into_make_service_with_connect_info::<SocketAddr>());
 
                     servers.push(Box::pin(s));
                 } else {
+                    let server = axum_server::Server::from_tcp(listener);
+
+                    let server = {
+                        server.acceptor(crate::sbd::SbdAcceptor::new(
```

Comment:
and probably for consistency would make sense to use that other custom acceptor before here as well even when not in tls mode